src/utils/math.ts

```diff
@@ -1,7 +1,9 @@
 export const clamp = (value: number, min: number, max: number): number => Math.max(min, Math.min(value, max));
 
-export const decimalPlaces = (value: number): number => {
-  const match = ("" + value).match(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/);
+const decimalNumberRegex = new RegExp(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/);
+
+export const decimalPlaces = (value: string): number => {
+  const match = value.match(decimalNumberRegex);
```

Comment:
Suggestion: would `test` help simplify this?

Reply:
We need to use the values of the match a couple lines down